src/game/TileEngine/Structure.cc

```diff
@@ -297,6 +297,43 @@ static void LoadStructureData(char const* const filename, STRUCTURE_FILE_REF* co
 	sfr->pubStructureData = structure_data.Release();
 }
 
+void NormalizeStructureTiles(DB_STRUCTURE_TILE** pTiles, UINT8 ubNumTiles)
+{
+	/**
+	 * In #1107, it was discovered that some of the Copter structures do not
+	 * have a base tile. RemoveStruct will not work without a base tile, as 
+	 * it tries to find a non-existent base structure.
+	 *
+	 * This function attempts to correct such issues with JSD data, by 
+	 * ensuring there is a tile with (0, 0) position relative to base.
+	 */
+	int minDistFromBase = INT_MAX;
+	for (UINT8 i = 0; i < ubNumTiles; i++)
+	{
+		DB_STRUCTURE_TILE* tile = pTiles[i];
+		if (abs(minDistFromBase) > abs(tile->sPosRelToBase))
+		{
+			minDistFromBase = tile->sPosRelToBase;
+		}
+	}
+
+	if (minDistFromBase == 0)
+	{
+		// Data is fine. Nothing to do.
+		return;
+	}
+
+	SLOGD(ST::format("Adjusting tiles relative positions by {}", -minDistFromBase));
+	int xDist = minDistFromBase % WORLD_COLS;
+	int yDist = minDistFromBase / WORLD_COLS;
```

Comment:
Shouldn't it be (I know they're the same currently):
```suggestion
	int yDist = minDistFromBase / WORLD_ROWS;
```

Reply:
I am not sure how these can converted - but I am following this: 

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/00111f7d7ad12438871434bc676d5b120e184d39/src/game/TileEngine/Structure.cc#L338

Reply:
Heh, that one looks wrong to me as well. Is it not trying to just compute the 1d index from 2d coordinates?

Reply:
I am not familiar on this enough to make such change. I'd rather keep it consistent with existing stuff.

Reply:
Nah, you're right, it's fine.